manager/orchestrator/jobs/orchestrator_controlapi_integration_test.go

```diff
@@ -0,0 +1,164 @@
+		By("setting up the controlapi client and server")
+		// a lot of this setup is taken from manager/controlapi/server_test.go,
+		// but it's unexported. in any case, re-writing it in ginkgo isn't
+		// unwarranted.
+		s = store.NewMemoryStore(&stateutils.MockProposer{})
+		server = controlapi.NewServer(s, nil, nil, nil, nil)
+
+		// we need a temporary unix socket to server on
+		temp, err := ioutil.TempFile("", "test-socket")
+		// i'm not sure why we immediately close this out, i'm not well versed
+		// in file system shenanigans, but this is what the controlapi unit
+		// tests do
+		Expect(err).ToNot(HaveOccurred())
+		Expect(temp.Close()).ToNot(HaveOccurred())
+		Expect(os.Remove(temp.Name())).ToNot(HaveOccurred())
+
+		tempUnixSocket = temp.Name()
+		lis, err := net.Listen("unix", temp.Name())
+		Expect(err).ToNot(HaveOccurred())
+
+		grpcServer = grpc.NewServer()
+		api.RegisterControlServer(grpcServer, server)
+
+		serverDone = testutils.EnsureRuns(func() {
+			_ = grpcServer.Serve(lis)
+		})
+
+		// the controlapi tests use the older grpc.Dial with a manually entered
+		// timeout. avoid that mess by just using the DialTimeout function
+		// instead.
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		// cancel after dial has completed is a no-op, but if we don't cancel,
+		// linters will (probably) complain about a leaked context.
+		defer cancel()
+		conn, err = grpc.DialContext(
+			ctx, "unix:"+temp.Name(),
+			// block on making this connection, to avoid the tests failing for
+			// funny reasons related to this connection being established async
+			grpc.WithBlock(),
+			grpc.WithInsecure(),
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		client = api.NewControlClient(conn)
+
+		By("setting up the orchestrator")
+		o = NewOrchestrator(s)
+		orchestratorDone = testutils.EnsureRuns(func() {
+			o.Run(context.Background())
+		})
+	})
+
+	AfterEach(func() {
+		conn.Close()
+		grpcServer.Stop()
+		// wait for the server to stop
+		<-serverDone
+		s.Close()
+		os.RemoveAll(tempUnixSocket)
+
+		o.Stop()
+		<-orchestratorDone
+	})
+
+	It("should create the requisite tasks for a new replicated job", func() {
+		spec := &api.ServiceSpec{
+			Annotations: api.Annotations{
+				Name: "testService",
+			},
+			Mode: &api.ServiceSpec_ReplicatedJob{
+				ReplicatedJob: &api.ReplicatedJob{
+					MaxConcurrent:    3,
+					TotalCompletions: 5,
+				},
+			},
+			Task: api.TaskSpec{
+				Runtime: &api.TaskSpec_Container{
+					Container: &api.ContainerSpec{
+						Image: "image",
+					},
+				},
+			},
+		}
+
+		resp, err := client.CreateService(context.Background(), &api.CreateServiceRequest{
+			Spec: spec,
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(resp).ToNot(BeNil())
+		Expect(resp.Service).ToNot(BeNil())
+
+		// Eventually is like Expect but it does polling, which is what we want
+		// If the function has two return values, this will additionally assert
+		// that the second value is nil
```

Comment:
TIL! I always do something like
```
var err error
Eventually(func() Whatever {
  foo, e := bar()
  if e != nil {
    err = e
  }
  ...
}
```
This is so much better!

Reply:
Yeah, I just happened to read this in the Ginkgo docs. It actually works for any number of arguments, not just 2, and will assert that any past the first are all nil.